src/components/ThemeEngine/ThemeEngine.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from 'react';
+
+import { kebabCase } from 'lodash';
+import type { Theme } from './ThemeEngine.types';
+import { CSS_PREFIX } from '../constants';
+import { DEFAULT_THEME_VARIANT, ThemeVariant } from './ThemeEngine.constants';
+import { themes } from './themes';
+
+export interface ThemeEngineProps {
+  theme?: ThemeVariant;
```

Comment:
i would make this property name a bit more descriptive. the type name is a good indicator that the property name isn't quite specific enough. it might not end up mattering. .. but it rarely hurts to be as specific as possible. in this case, it is likely not going to be very long before themes are more than just light/dark. it will likely be a combination light/dark defaults and custom overrides. while we don't want to make too many assumptions about how something will end up looking, we also don't want to prematurely limit ourselves either. i think naming it `variant` or `mode` or `themeVariant` would be more specific, and reduce the possibility of a future conflict.